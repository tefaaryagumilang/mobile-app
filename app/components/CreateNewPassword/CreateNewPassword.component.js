import React from 'react';
import PropTypes from 'prop-types';
import {View, Text, Image} from 'react-native';
import styles from './CreateNewPassword.styles.js';
import {Field} from 'redux-form';
import Touchable from '../Touchable.component';
import {SinarmasInput, SinarmasButton} from '../FormComponents';
import noop from 'lodash/noop';
import {language} from '../../config/language';
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view';
import showPasswordIcon from '../../assets/images/blackEyeIcon.png';
import hidePasswordIcon from '../../assets/images/blackEyeSlashIcon.png';

class CreateNewPassword extends React.Component {
  static propTypes = {
    submitting: PropTypes.bool,
    invalid: PropTypes.bool,
    handleSubmit: PropTypes.func,
    showOrHideNewPassword: PropTypes.func,
    isSecureTextEntryNewPassword: PropTypes.bool,
    showOrHideConfirmNewPassword: PropTypes.func,
    isSecureTextEntryConfirmNewPassword: PropTypes.bool,
    oldPassword: PropTypes.string,
    userName: PropTypes.string,
    validationInput: PropTypes.func,
    setNewPasswordValue: PropTypes.func,
    passwordValue: PropTypes.string
  }

  render () {
    const {submitting, invalid, showOrHideNewPassword, isSecureTextEntryNewPassword = true, showOrHideConfirmNewPassword, isSecureTextEntryConfirmNewPassword = true, handleSubmit = noop, oldPassword = '', userName = '', passwordValue = '', validationInput, setNewPasswordValue} = this.props;
    const createNewPassword = language.BUTTON__NEXT + ' ' + language.CHANGE_PASSWORD__CONFIRM_NEW_PASSWORD;
    return (
      <KeyboardAwareScrollView keyboardShouldPersistTaps='handled' contentContainerStyle={styles.container} extraHeight={120}>
        <View>
          <View style={styles.inputSpacing}>
            <View style={styles.passwordFieldsContainer}>
              <Field
                name='newPassword'
                label={language.CHANGE_PASSWORD__ENTER_NEW_PASSWORD}
                placeholder={language.HINTTEXT__PASSWORD}
                component={SinarmasInput}
                secureTextEntry={isSecureTextEntryNewPassword}
                isUseSuccessInputText={true}
                typeField={'newPassword'}
                usernameValue={userName}
                oldPassword={oldPassword}
                validationInput={validationInput}
                setNewPasswordValue={setNewPasswordValue}
              />
              <Touchable onPress={showOrHideNewPassword} style={styles.eyeIconStyle}><Image source={isSecureTextEntryNewPassword ? showPasswordIcon : hidePasswordIcon}/></Touchable>
            </View>
            <Text style={styles.passRule}>{language.ONBOARDING__ACCOUNT_PASSWORD_SUBTEXT}</Text>
          </View>
          <View style={styles.inputSpacing}>
            <View style={styles.passwordFieldsContainer}>
              <Field
                name='confirmNewPassword'
                label={language.CHANGE_PASSWORD__CONFIRM_NEW_PASSWORD}
                placeholder={language.HINTTEXT__PASSWORD}
                component={SinarmasInput}
                secureTextEntry={isSecureTextEntryConfirmNewPassword}
                isUseSuccessInputText={true}
                typeField={'confirmNewPassword'}
                usernameValue={userName}
                oldPassword={oldPassword}
                passwordValue={passwordValue}
                validationInput={validationInput}
              />
              <Touchable onPress={showOrHideConfirmNewPassword} style={styles.eyeIconStyle}><Image source={isSecureTextEntryConfirmNewPassword ? showPasswordIcon : hidePasswordIcon}/></Touchable>
            </View>
          </View>
        </View>
        <SinarmasButton dtActionName = {createNewPassword} text={language.BUTTON__NEXT} onPress={handleSubmit} disabled={(submitting || invalid)}/>
      </KeyboardAwareScrollView>
    );
  }
}

export default CreateNewPassword;
