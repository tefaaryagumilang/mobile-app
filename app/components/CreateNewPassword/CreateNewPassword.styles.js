import {contentContainerStyle, fontSizeNormalStyle, fontSizeSmallStyle, textLightGreyStyle} from '../../styles/common.styles';

export default {
  container: [contentContainerStyle, {
    justifyContent: 'space-between',
    flex: 1
  }],
  label: [fontSizeNormalStyle],
  inputSpacing: {
    marginBottom: 15
  },
  passwordFieldsContainer: {
    flexDirection: 'column'
  },
  eyeIconStyle: {position: 'absolute', width: 30, right: 5, top: 30},
  passRule: [fontSizeSmallStyle, textLightGreyStyle]
};
