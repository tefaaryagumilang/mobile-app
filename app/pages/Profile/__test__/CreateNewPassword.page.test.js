import React from 'react';
import renderer from 'react-test-renderer';
import CreateNewPassword from '../CreateNewPassword.page';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {reduxForm} from 'redux-form';

const spy = jest.fn();
const store = createStore(() => ({}));
const DecoratedForm = reduxForm({form: 'testForm'})(CreateNewPassword);

describe('Profile: CreateNewPassword page', () => {
  xit('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <DecoratedForm handleSubmit={spy}/>
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
