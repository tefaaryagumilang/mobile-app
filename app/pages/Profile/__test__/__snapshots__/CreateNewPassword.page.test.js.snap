// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile: CreateNewPassword page renders correctly 1`] = `
<KeyboardAwareScrollView
  contentContainerStyle={
    Array [
      Object {
        "padding": 20,
      },
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      },
    ]
  }
  extraHeight={120}
  keyboardShouldPersistTaps="handled"
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 14,
            },
          ]
        }
      >
        masukkan kata sandi baru
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#ededed",
                  "marginTop": 10,
                },
                Object {},
              ]
            }
          >
            <TextInput
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor={undefined}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 5,
              "top": 25,
              "width": 30,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 12,
            },
            Object {
              "color": "#666",
            },
          ]
        }
      >
        Minimal 8 karakter. Alfanumerik. Gunakan minimal 1 huruf besar.
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontSize": 14,
            },
          ]
        }
      >
        Konfirmasi kata sandi baru
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#ededed",
                  "marginTop": 10,
                },
                Object {},
              ]
            }
          >
            <TextInput
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholderTextColor={undefined}
              secureTextEntry={true}
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 16,
                    "fontWeight": "300",
                    "height": 50,
                    "padding": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "position": "absolute",
              "right": 5,
              "top": 25,
              "width": 30,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
          />
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ED1D25",
            "borderRadius": 30,
            "flexDirection": "row",
            "height": 47,
            "justifyContent": "center",
            "paddingHorizontal": 15,
          },
          undefined,
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 16,
          }
        }
      >
        Berikut
      </Text>
    </View>
  </View>
</KeyboardAwareScrollView>
`;
